src/plugins/intel_gna/transformations/insert_copy_layer.cpp

```diff
@@ -0,0 +1,358 @@
+NGRAPH_RTTI_DEFINITION(InsertCopyBeforeConcatLayer, "InsertCopyBeforeConcatLayer", 0);
+NGRAPH_RTTI_DEFINITION(HandleMultiConnectedLayerToConcatAndMemory, "HandleMultiConnectedLayerToConcatAndMemory", 0);
+NGRAPH_RTTI_DEFINITION(MatchNonComputationalLayers, "MatchNonComputationalLayers", 0);
+NGRAPH_RTTI_DEFINITION(HandleNonComputationalSubgraphs, "HandleNonComputationalSubgraphs", 0);
+
+
+namespace {
+
+    bool is_aligned_split(std::shared_ptr<ngraph::Node> input_op, size_t input_op_out_index) {
+        size_t offset = 0;
+        // currently split layer only supports 2 bytes in int16 and int8 mode. In fp32 mode this is not necessary but is useful for testing
+        const int bytesPerSplitElement = 2;
+
+        if (std::dynamic_pointer_cast<ngraph::opset8::Split>(input_op) || std::dynamic_pointer_cast<ngraph::opset8::VariadicSplit>(input_op)) {
+            for (size_t index = 0; index < input_op_out_index; index++) {
+                auto curr_out_shape = input_op->get_output_shape(index);
+                auto outputSize = InferenceEngine::details::product(std::begin(curr_out_shape), std::end(curr_out_shape));
+                offset += outputSize * bytesPerSplitElement;
+            }
+        }
+        return offset == ALIGN64(offset);
+    }
+
+    void insert_copy_layer_between(std::shared_ptr<ngraph::Node> input_op,
+                                   std::shared_ptr<ngraph::Node> output_op,
+                                   const size_t& index) {
+        NGRAPH_CHECK(input_op);
+        NGRAPH_CHECK(output_op);
+
+        auto input_op_out_index = output_op->input(index).get_source_output().get_index();
+        // In this case we no need copy layer insertion, because after insertion of aligning filter graph will have conv func layer
+        // Should be removed, when InsertSplitAligningFilterPass will be moved on nGraph, because it should be running before the copy layer passes
+        if (!is_aligned_split(input_op, input_op_out_index))
+            return;
+
+        auto copy_op = std::make_shared<ov::intel_gna::op::Copy>(input_op->output(input_op_out_index));
+        copy_op->set_friendly_name(input_op->get_friendly_name() + "/copy_layer/" + output_op->get_friendly_name() + "." + std::to_string(index));
+        ngraph::copy_runtime_info(input_op, copy_op);
+
+        output_op->input(index).replace_source_output(copy_op);
+    }
+
+    bool is_crop_affined(std::shared_ptr<ngraph::Node> node) {
+        auto crop = std::dynamic_pointer_cast<ngraph::op::CropIE>(node);
+        if (crop != nullptr && !crop->offset.empty()) {
+            // currently crop layer only supports 2 bytes in int16 and int8 mode.
+            // In fp32 mode this is not necessary but is useful for testing
+            size_t bytesPerCropElement = 2;
+            size_t cropOffset = crop->offset.back() * bytesPerCropElement;
+            return (ALIGN64(cropOffset) != cropOffset);
+        }
+        return false;
+    }
```

Comment:
IMO would be better to reuse the logic from `gna_layer_info.hpp` instead of duplicating very similar logic.

Reply:
> IMO would be better to reuse the logic from `gna_layer_info.hpp` instead of duplicating very similar logic.

I created common variable `bytesPerCropElement` for both functions - legacy one and ngraph based. I don't know what also can be reused here, because we are working with different datat types here.

Reply:
eg, such function could be shared:
bool is_crop_affined_offset(size_t numberOfElements) {
     const auto cropOffset = numberOfElements* bytesPerCropElement;
     return (ALIGN64(cropOffset) != cropOffset);
}